File: RemoveDuplicates.cs
//O(n)
//O(1)
public class Solution
{
    public int RemoveDuplicates(int[] nums)
    {
        if (nums.Length == 0) return 0;

        int count = 1;
        for (int i = 1; i < nums.Length; i++)
        {
            if (nums[i] != nums[i - 1])
            {
                nums[count] = nums[i];
                count++;
            }
        }

        return count;
    }

}

File: C++/removeDuplicates.cpp
// O(N)
// O(1)
#include <vector>
class Solution {
public:
    int removeDuplicates(std::vector<int>& nums) {
        if (nums.empty()) return 0;

        int result = 1;

        for(int i = 0; i < nums.size(); i++)
        {
            if(nums[i] != nums[result -1])
            {
                nums[result++] = nums[i];
            }
        }

        return result;
    }
};